backends/oauth2_test.go

```diff
@@ -0,0 +1,303 @@
+package backends
+
+import (
+	log "github.com/sirupsen/logrus"
+	. "github.com/smartystreets/goconvey/convey"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+	"time"
+)
+
+func setupMockOAuthServer() (*httptest.Server, func()) {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/userinfo", func(w http.ResponseWriter, r *http.Request) {
+		authHeader := r.Header.Get("Authorization")
+
+		if authHeader == "" || authHeader == "Bearer wrong_token" {
+			http.Error(w, "Fail", 404)
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+
+		if authHeader == "Bearer mock_access_token_normaluser" {
+			w.Write([]byte("{\"sub\":\"mock_user_id_0\",\"mqtt\":{\"superuser\":false,\"topics\":{\"read\":[\"/test/topic/read/#\",\"/test/topic/writeread/1\",\"/test/topic/pattern/username/%u\",\"/test/topic/pattern/clientid/%c\"],\"write\":[\"/test/topic/write/+/db\",\"/test/topic/writeread/1\"]}}}"))
```

Comment:
The line length is quite big here and in other places. Maybe these could be properly defined structs and marshalled before writing them to the `ResponseWriter`, improving readability as well as making it easier to modify them later.

Reply:
i realized your recommendation in [commit](https://github.com/iegomez/mosquitto-go-auth/pull/210/commits/389a297c008172b316e08d05f6187c3a87af4700)